### ISSUE TITLE
[RyuJIT/ARM32] Unhandled Exception: System.FormatException: argument index

### ISSUE THREAD

**Issue by hseok-oh** [dotnet/coreclr#12702]
Unhandled Exception: System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.

```
Unhandled Exception: System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
     at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
     at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
     at System.String.Format(IFormatProvider provider, String format, Object arg0)
     at System.IO.TextWriter.WriteLine(String format, Object arg0)
.....
```

<details><summary>159 tests</summary>

```
GC/API/GC/AddMemoryPressureTest/AddMemoryPressureTest.sh
GC/API/GC/AddUsageTest/AddUsageTest.sh
GC/API/GC/GetGeneration_fail/GetGeneration_fail.sh
GC/API/GC/GetTotalMemory/GetTotalMemory.sh
GC/API/GC/KeepAliveRecur/KeepAliveRecur.sh
GC/API/GC/RemoveUsageTest/RemoveUsageTest.sh
GC/API/GCSettings/InputValidation/InputValidation.sh
GC/Coverage/LargeObjectAlloc/LargeObjectAlloc.sh
GC/Features/Pinning/PinningOther/PinnedCollect/PinnedCollect.sh
GC/Features/Pinning/PinningOther/PinnedObject/PinnedObject.sh
GC/Scenarios/RanCollect/rancollect/rancollect.sh
GC/Scenarios/Samples/gc/gc.sh
GC/Scenarios/THDChaos/thdchaos/thdchaos.sh
GC/Scenarios/THDList/thdlist/thdlist.sh
JIT/Directed/StructPromote/SP2a/SP2a.sh
JIT/Directed/coverage/oldtests/cse2_cs_d/cse2_cs_d.sh
JIT/Directed/coverage/oldtests/cse2_cs_do/cse2_cs_do.sh
JIT/Directed/coverage/oldtests/cse2_cs_r/cse2_cs_r.sh
JIT/Directed/coverage/oldtests/cse2_cs_ro/cse2_cs_ro.sh
JIT/Directed/shift/int64_d/int64_d.sh
JIT/Directed/shift/int64_do/int64_do.sh
JIT/Directed/shift/int64_r/int64_r.sh
JIT/Directed/shift/int64_ro/int64_ro.sh
JIT/Directed/shift/uint64_d/uint64_d.sh
JIT/Directed/shift/uint64_do/uint64_do.sh
JIT/Directed/shift/uint64_r/uint64_r.sh
JIT/Directed/shift/uint64_ro/uint64_ro.sh
JIT/Generics/Conversions/Boxing/box_unbox01/box_unbox01.sh
JIT/Generics/Typeof/objectBoxing/objectBoxing.sh
JIT/Generics/Typeof/refTypesdynamic/refTypesdynamic.sh
JIT/Generics/Typeof/valueTypeBoxing/valueTypeBoxing.sh
JIT/Methodical/VT/callconv/_il_reljumper4/_il_reljumper4.sh
JIT/Methodical/eh/cs/unsafe_do/unsafe_do.sh
JIT/Methodical/eh/interactions/gcincatch_do/gcincatch_do.sh
JIT/Methodical/eh/interactions/gcincatch_r/gcincatch_r.sh
JIT/Methodical/eh/interactions/gcincatch_ro/gcincatch_ro.sh
JIT/Methodical/eh/nested/nonlocalexit/throwinfinallynestedintry_30_do/throwinfinallynestedintry_30_do.sh
JIT/Methodical/flowgraph/bug619534/finallyclone/finallyclone.sh
JIT/Methodical/flowgraph/dev10_bug642944/GCMaskForGSCookie/GCMaskForGSCookie.sh
JIT/Methodical/fp/apps/bouncingball_cs_d/bouncingball_cs_d.sh
JIT/Methodical/fp/apps/bouncingball_cs_do/bouncingball_cs_do.sh
JIT/Methodical/fp/apps/bouncingball_cs_r/bouncingball_cs_r.sh
JIT/Methodical/fp/apps/bouncingball_cs_ro/bouncingball_cs_ro.sh
JIT/Methodical/int64/signed/_dbgs_ldc_div/_dbgs_ldc_div.sh
JIT/Methodical/int64/signed/_dbgs_ldc_mul/_dbgs_ldc_mul.sh
JIT/Methodical/int64/signed/_dbgs_ldc_mulovf/_dbgs_ldc_mulovf.sh
JIT/Methodical/int64/signed/_dbgs_ldfld_mul/_dbgs_ldfld_mul.sh
JIT/Methodical/int64/signed/_dbgs_ldfld_mulovf/_dbgs_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_dbgs_ldsfld_mul/_dbgs_ldsfld_mul.sh
JIT/Methodical/int64/signed/_dbgs_ldsfld_mulovf/_dbgs_ldsfld_mulovf.sh
JIT/Methodical/int64/signed/_il_dbgs_ldc_div/_il_dbgs_ldc_div.sh
JIT/Methodical/int64/signed/_il_dbgs_ldc_mul/_il_dbgs_ldc_mul.sh
JIT/Methodical/int64/signed/_il_dbgs_ldc_mulovf/_il_dbgs_ldc_mulovf.sh
JIT/Methodical/int64/signed/_il_dbgs_ldfld_mul/_il_dbgs_ldfld_mul.sh
JIT/Methodical/int64/signed/_il_dbgs_ldfld_mulovf/_il_dbgs_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_il_dbgs_ldsfld_mul/_il_dbgs_ldsfld_mul.sh
JIT/Methodical/int64/signed/_il_dbgs_ldsfld_mulovf/_il_dbgs_ldsfld_mulovf.sh
JIT/Methodical/int64/signed/_il_rels_ldc_div/_il_rels_ldc_div.sh
JIT/Methodical/int64/signed/_il_rels_ldc_mul/_il_rels_ldc_mul.sh
JIT/Methodical/int64/signed/_il_rels_ldc_mulovf/_il_rels_ldc_mulovf.sh
JIT/Methodical/int64/signed/_il_rels_ldfld_mul/_il_rels_ldfld_mul.sh
JIT/Methodical/int64/signed/_il_rels_ldfld_mulovf/_il_rels_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_il_rels_ldsfld_mul/_il_rels_ldsfld_mul.sh
JIT/Methodical/int64/signed/_il_rels_ldsfld_mulovf/_il_rels_ldsfld_mulovf.sh
JIT/Methodical/int64/signed/_rels_ldc_div/_rels_ldc_div.sh
JIT/Methodical/int64/signed/_rels_ldc_mul/_rels_ldc_mul.sh
JIT/Methodical/int64/signed/_rels_ldc_mulovf/_rels_ldc_mulovf.sh
JIT/Methodical/int64/signed/_rels_ldfld_mul/_rels_ldfld_mul.sh
JIT/Methodical/int64/signed/_rels_ldfld_mulovf/_rels_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_rels_ldsfld_mul/_rels_ldsfld_mul.sh
JIT/Methodical/int64/signed/_rels_ldsfld_mulovf/_rels_ldsfld_mulovf.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldc_div/_speed_dbgs_ldc_div.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldc_mul/_speed_dbgs_ldc_mul.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldc_mulovf/_speed_dbgs_ldc_mulovf.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldfld_mul/_speed_dbgs_ldfld_mul.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldfld_mulovf/_speed_dbgs_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldsfld_mul/_speed_dbgs_ldsfld_mul.sh
JIT/Methodical/int64/signed/_speed_dbgs_ldsfld_mulovf/_speed_dbgs_ldsfld_mulovf.sh
JIT/Methodical/int64/signed/_speed_rels_ldc_div/_speed_rels_ldc_div.sh
JIT/Methodical/int64/signed/_speed_rels_ldc_mul/_speed_rels_ldc_mul.sh
JIT/Methodical/int64/signed/_speed_rels_ldc_mulovf/_speed_rels_ldc_mulovf.sh
JIT/Methodical/int64/signed/_speed_rels_ldfld_mul/_speed_rels_ldfld_mul.sh
JIT/Methodical/int64/signed/_speed_rels_ldfld_mulovf/_speed_rels_ldfld_mulovf.sh
JIT/Methodical/int64/signed/_speed_rels_ldsfld_mul/_speed_rels_ldsfld_mul.sh
JIT/Methodical/int64/signed/_speed_rels_ldsfld_mulovf/_speed_rels_ldsfld_mulovf.sh
JIT/Methodical/int64/superlong/_dbgsuperlong/_dbgsuperlong.sh
JIT/Methodical/int64/unsigned/_dbgldc_mulovf/_dbgldc_mulovf.sh
JIT/Methodical/int64/unsigned/_dbgldfld_mul/_dbgldfld_mul.sh
JIT/Methodical/int64/unsigned/_dbgldfld_mulovf/_dbgldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_dbgldsfld_mul/_dbgldsfld_mul.sh
JIT/Methodical/int64/unsigned/_dbgldsfld_mulovf/_dbgldsfld_mulovf.sh
JIT/Methodical/int64/unsigned/_il_dbgldc_mul/_il_dbgldc_mul.sh
JIT/Methodical/int64/unsigned/_il_dbgldc_mulovf/_il_dbgldc_mulovf.sh
JIT/Methodical/int64/unsigned/_il_dbgldfld_mul/_il_dbgldfld_mul.sh
JIT/Methodical/int64/unsigned/_il_dbgldfld_mulovf/_il_dbgldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_il_dbgldsfld_mul/_il_dbgldsfld_mul.sh
JIT/Methodical/int64/unsigned/_il_dbgldsfld_mulovf/_il_dbgldsfld_mulovf.sh
JIT/Methodical/int64/unsigned/_il_relldc_mul/_il_relldc_mul.sh
JIT/Methodical/int64/unsigned/_il_relldc_mulovf/_il_relldc_mulovf.sh
JIT/Methodical/int64/unsigned/_il_relldfld_mul/_il_relldfld_mul.sh
JIT/Methodical/int64/unsigned/_il_relldfld_mulovf/_il_relldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_il_relldsfld_mul/_il_relldsfld_mul.sh
JIT/Methodical/int64/unsigned/_il_relldsfld_mulovf/_il_relldsfld_mulovf.sh
JIT/Methodical/int64/unsigned/_relldc_mul/_relldc_mul.sh
JIT/Methodical/int64/unsigned/_relldc_mulovf/_relldc_mulovf.sh
JIT/Methodical/int64/unsigned/_relldfld_mul/_relldfld_mul.sh
JIT/Methodical/int64/unsigned/_relldfld_mulovf/_relldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_relldsfld_mul/_relldsfld_mul.sh
JIT/Methodical/int64/unsigned/_relldsfld_mulovf/_relldsfld_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_dbgldc_mul/_speed_dbgldc_mul.sh
JIT/Methodical/int64/unsigned/_speed_dbgldc_mulovf/_speed_dbgldc_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_dbgldfld_mul/_speed_dbgldfld_mul.sh
JIT/Methodical/int64/unsigned/_speed_dbgldfld_mulovf/_speed_dbgldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_dbgldsfld_mul/_speed_dbgldsfld_mul.sh
JIT/Methodical/int64/unsigned/_speed_dbgldsfld_mulovf/_speed_dbgldsfld_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_relldc_mul/_speed_relldc_mul.sh
JIT/Methodical/int64/unsigned/_speed_relldc_mulovf/_speed_relldc_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_relldfld_mul/_speed_relldfld_mul.sh
JIT/Methodical/int64/unsigned/_speed_relldfld_mulovf/_speed_relldfld_mulovf.sh
JIT/Methodical/int64/unsigned/_speed_relldsfld_mul/_speed_relldsfld_mul.sh
JIT/Methodical/int64/unsigned/_speed_relldsfld_mulovf/_speed_relldsfld_mulovf.sh
JIT/Methodical/switch/switch6/switch6.sh
JIT/Performance/CodeQuality/BenchmarksGame/binarytrees/binarytrees/binarytrees.sh
JIT/Performance/CodeQuality/BenchmarksGame/binarytrees/binarytrees3/binarytrees3.sh
JIT/Performance/CodeQuality/BenchmarksGame/spectralnorm/spectralnorm/spectralnorm.sh
JIT/Performance/CodeQuality/SciMark/SciMark/SciMark.sh
JIT/Performance/CodeQuality/Span/Indexer/Indexer.sh
JIT/Regression/CLR-x86-JIT/V1-M09.5-PDC/b10665/b10665/b10665.sh
JIT/Regression/CLR-x86-JIT/V1-M09.5-PDC/b10666/b10666/b10666.sh
JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b74182/b74182/b74182.sh
JIT/Regression/CLR-x86-JIT/V1-M13-RTM/b87284/b87284/b87284.sh
JIT/Regression/CLR-x86-JIT/V2.0-Beta2/b321799/b321799/b321799.sh
JIT/Regression/CLR-x86-JIT/V2.0-Beta2/b441487/b441487/b441487.sh
JIT/Regression/CLR-x86-JIT/v2.1/b609988/Desktop/b609988/b609988.sh
JIT/Regression/CLR-x86-JIT/v2.1/b609988/b609988/b609988.sh
JIT/Regression/Dev11/Dev11_457559/Dev11_457559/Dev11_457559.sh
JIT/Regression/Dev11/External/Dev11_243742/app/app.sh
JIT/Regression/Dev11/External/dev11_154899/DynamicStaticAlignment/DynamicStaticAlignment.sh
JIT/Regression/Dev11/dev11_165544/seqpts/seqpts.sh
JIT/Regression/VS-ia64-JIT/M00/b141062/b141062/b141062.sh
JIT/Regression/VS-ia64-JIT/V1.2-Beta1/b124409/b124409/b124409.sh
JIT/Regression/VS-ia64-JIT/V1.2-M02/b10828/b10828/b10828.sh
JIT/SIMD/VectorAbs_r/VectorAbs_r.sh
JIT/SIMD/VectorAbs_ro/VectorAbs_ro.sh
JIT/SIMD/VectorAdd_r/VectorAdd_r.sh
JIT/SIMD/VectorAdd_ro/VectorAdd_ro.sh
JIT/SIMD/VectorArgs_r/VectorArgs_r.sh
JIT/SIMD/VectorArray_r/VectorArray_r.sh
JIT/SIMD/VectorCast_ro/VectorCast_ro.sh
JIT/SIMD/VectorDiv_r/VectorDiv_r.sh
JIT/SIMD/VectorMax_ro/VectorMax_ro.sh
JIT/SIMD/VectorMin_r/VectorMin_r.sh
JIT/SIMD/VectorMul_r/VectorMul_r.sh
JIT/SIMD/VectorMul_ro/VectorMul_ro.sh
JIT/SIMD/VectorRelOp_ro/VectorRelOp_ro.sh
JIT/SIMD/VectorSub_r/VectorSub_r.sh
JIT/jit64/opt/cse/HugeField1/HugeField1.sh
JIT/jit64/regress/vsw/373472/test_il/test_il.sh
JIT/opt/perf/doublealign/objects/objects.sh
```
</details>

Unhandled Exception: System.TypeInitializationException: The type initializer for XXX threw an exception. 
---> (....)
---> System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.

<details><summary>16 tests</summary>

```
JIT/Regression/CLR-x86-JIT/v2.1/b565808/b565808/b565808.sh
Loader/classloader/generics/Statics/Regressions/524571/StaticsProblem1/StaticsProblem1.sh
Loader/classloader/generics/Statics/Regressions/524571/StaticsProblem3/StaticsProblem3.sh
Loader/classloader/generics/Statics/Regressions/524571/StaticsProblem4/StaticsProblem4.sh
Loader/classloader/generics/Statics/Regressions/524571/StaticsProblem5/StaticsProblem5.sh
Loader/classloader/generics/regressions/vsw524571/StaticsProblem5/StaticsProblem5.sh
Loader/classloader/generics/Statics/Regressions/524571/StaticsProblem2/StaticsProblem2.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorFourThreadsBFI/CircularCctorFourThreadsBFI.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads01BFI/CircularCctorThreeThreads01BFI.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads02BFI/CircularCctorThreeThreads02BFI.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads03BFI/CircularCctorThreeThreads03BFI.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorFourThreads/CircularCctorFourThreads.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads01/CircularCctorThreeThreads01.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads02/CircularCctorThreeThreads02.sh
Loader/classloader/TypeInitialization/CircularCctors/CircularCctorThreeThreads03/CircularCctorThreeThreads03.sh
Loader/classloader/TypeInitialization/CoreCLR/CircularCctorThreeThreads03/CircularCctorThreeThreads03.sh
```
</details>

**Comment by hseok-oh** [comment]
It seems related to WriteLine method (or handling varargs),

**Comment by hseok-oh** [comment]
Fixed by #12710
